### PR TITLE
volume persistence deletion tests

### PR DESCRIPTION
**Pull request by heavenlyhash**
Noticed some issues with consistency around deletion; fixed and added missing tests.

n.b. This should have gone in with #973 -- I'd be happy to squash it into that branch after a quick review if desired.

### CODE REVIEW

File: host/volume/manager/manager.go [hunk 1]
```diff
@@ -279,12 +296,12 @@ func (m *Manager) getProviderBucket(tx *bolt.Tx, providerID string) (*bolt.Bucke
...
-func (m *Manager) persistVolume(tx *bolt.Tx, id string) error {
+// Called to sync changes to disk when a volume is updated
+func (m *Manager) persistVolume(tx *bolt.Tx, vol volume.Volume) error {
 	// Save the general volume info
 	volumesBucket := tx.Bucket([]byte("volumes"))
-	k := []byte(id)
-	vol, ok := m.volumes[id]
+	k := []byte(vol.Info().ID)
+	_, ok := m.volumes[vol.Info().ID]
 	if !ok {
```
Comment (titanous) [inline host/volume/manager/manager.go]: Could merge the ok check into this if statement

File: host/volume/manager/manager.go [hunk 2]
```diff
@@ -279,12 +296,12 @@ func (m *Manager) getProviderBucket(tx *bolt.Tx, providerID string) (*bolt.Bucke
...
 
-// Called to sync changes to disk when a VolumeInfo is updated
-func (m *Manager) persistVolume(tx *bolt.Tx, id string) error {
+// Called to sync changes to disk when a volume is updated
+func (m *Manager) persistVolume(tx *bolt.Tx, vol volume.Volume) error {
 	// Save the general volume info
 	volumesBucket := tx.Bucket([]byte("volumes"))
-	k := []byte(id)
-	vol, ok := m.volumes[id]
+	k := []byte(vol.Info().ID)
```
Comment (titanous) [inline host/volume/manager/manager.go]: How expensive is calling `Info`?

Comment (heavenlyhash) [inline host/volume/manager/manager.go]: somewhere in the area of a vtable jump, but if you're suggesting this reads like a stutter I'd have to concede; will replace.

Comment (titanous) [inline host/volume/manager/manager.go]: Yeah, not a big deal, but I'd use a temp as it's called three times.

File: host/volume/manager/manager.go (hunk 1 above)
Comment (heavenlyhash) [inline host/volume/manager/manager.go]: It's used again further down in the branch for whether or not the provider storage is doing a deletion.  I'll rename so we don't have an 'ok' floating around that long.

**Comment by heavenlyhash** [comment]
Comments addressed

**Comment by titanous** [comment]
LGTM

**Pull request merged**